compiler/src/org.graalvm.compiler.java/src/org/graalvm/compiler/java/BytecodeParser.java

```diff
@@ -1227,7 +1227,30 @@ protected void handleUnresolvedLoadConstant(JavaType type) {
      */
     protected void handleUnresolvedCheckCast(JavaType type, ValueNode object) {
         assert !graphBuilderConfig.unresolvedIsError();
-        append(new FixedGuardNode(graph.addOrUniqueWithInputs(IsNullNode.create(object)), Unresolved, InvalidateRecompile));
```

Comment:
Was this change necessary? Or was it just for symmetry with the SVM code?

This structure will just end up being turned back into the `FixedGuardNode` so i think we should just keep this as is. Unless it fixes some issue i'm missing.

Reply:
> Was this change necessary? Or was it just for symmetry with the SVM code?

Only for symmetry.

> This structure will just end up being turned back into the FixedGuardNode so i think we should just keep this as is. Unless it fixes some issue i'm missing.

I'll try that instead.